crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -400,15 +400,12 @@ impl Display for SuiParsedTransactionResponse {
 
 #[allow(clippy::large_enum_variant)]
 #[derive(Serialize, Deserialize, Debug, JsonSchema)]
-pub enum SuiExecuteTransactionResponse {
-    // TODO: Change to CertifiedTransactionEffects eventually.
-    EffectsCert {
-        certificate: SuiCertifiedTransaction,
-        effects: SuiCertifiedTransactionEffects,
-        // If the transaction is confirmed to be executed locally
-        // before this response.
-        confirmed_local_execution: bool,
-    },
+pub struct SuiExecuteTransactionResponse {
+    pub certificate: SuiCertifiedTransaction,
+    pub effects: SuiCertifiedTransactionEffects,
+    // If the transaction is confirmed to be executed locally
+    // before this response.
+    pub confirmed_local_execution: bool,
 }
```

Comment:
To make the TS-SDK backward compatible with this change: we would need to mark the [existing TS type](https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/types/transactions.ts#L229-L241) as deprecated and add a new one, and return the right type in https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/providers/json-rpc-provider.ts#L578-L609 based on the [RPC version](https://github.com/MystenLabs/sui/blob/8474242afcab175830b64817821a37a497edb3a1/sdk/typescript/src/providers/json-rpc-provider.ts#L147)

Reply:
thanks, will follow the instructions to make the changes

Reply:
https://github.com/MystenLabs/sui/pull/7468/files#r1091322788